Build failure: openmsx

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

1. Create a file named `openmsx.nix` that contains this Nix expression:

    ```nix
    # This file is dedicated to the public domain using 🅭🄍1.0:
    # <https://creativecommons.org/publicdomain/zero/1.0>.
    let
      nixpkgsCommit = "6c6c2229cc86932f732faea2784231ab9645dcaa";
      nixpkgsTarball = builtins.fetchTarball {
        url = "https://github.com/NixOS/nixpkgs/archive/${nixpkgsCommit}.tar.gz";
        sha256 = "0bzy7mmaycv6ib1b2njy8bs9ymvkg4rnfy9lyh6qppnnlqxg24lw";
      };
      pkgs = import nixpkgsTarball { };
    in
    pkgs.openmsx
    ```

1. Try to build that Nix expression by running this command:

    ```bash
    nix-build <path to openmsx.nix>
    ```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/302614089

### Relevant log output

```console
$ cat Log.txt 
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/zv7pn7ikxbbh3ayigdc75hjlwj20j11p-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
configure flags: ''
Using Python: python3
Probing target system...
Creating derived/x86_64-linux-opt/config/probed_defs.mk...
Creating derived/x86_64-linux-opt/config/systemfuncs.hh...

Found libraries:
  ALSA:             version 1.2.14
  FreeType:         version 
  GLEW:             version unknown
  libogg:           version unknown
  libpng:           version 1.6.47
  libtheora:        version unknown
  libvorbis:        version unknown
  OpenGL:           version 4.6
  SDL2:             version 2.32.56
  SDL2_ttf:         version 2.24.0
  Tcl:              version 8.6.15
  zlib:             version 1.3.1

Components overview:
  Emulation core:   yes
  GL renderer:      yes
  Laserdisc:        yes
  ALSA MIDI:        yes

Customisable options:
  Install to        ${out}
  (you can edit these in build/custom.mk)

All required and optional components can be built.

If the detected libraries differ from what you think is installed on this system, please check the log file: derived/x86_64-linux-opt/config/probe.log

Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash
Using Python: python3
Creating derived/x86_64-linux-opt/config/components_defs.mk...
Using Python: python3
Build configuration:
  Platform: x86_64-linux
  Flavour:  opt
  Compiler: g++
  Subset:   full build
Creating derived/x86_64-linux-opt/config/Version.ii...
Creating derived/x86_64-linux-opt/config/build-info.hh...
Creating derived/x86_64-linux-opt/config/components.hh...
Compiling Autofire.cc...
Compiling CLIOption.cc...
Compiling CartridgeSlotManager.cc...
Compiling ChakkariCopy.cc...
Compiling CliExtension.cc...
Compiling CommandLineParser.cc...
In file included from src/settings/EnumSetting.hh:5,
                 from src/CommandLineParser.cc:14:
src/utils/view.hh: In instantiation of 'constexpr view::detail::TransformIterator<hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >::difference_type view::detail::operator-(const TransformIterator<hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >&, const TransformIterator<hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >&)':
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/ranges_base.h:961:23:   required from 'constexpr std::iter_difference_t<typename std::decay<_Tp>::type> std::ranges::__distance_fn::operator()(_It&&, _Sent) const [with _It = view::detail::TransformIterator<hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >&; _Sent = view::detail::TransformIterator<hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >; std::iter_difference_t<typename std::decay<_Tp>::type> = int; typename std::decay<_Tp>::type = view::detail::TransformIterator<hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >]'
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/stl_vector.h:718:44:   required from 'constexpr std::vector<_Tp, _Alloc>::vector(_InputIterator, _InputIterator, const allocator_type&) [with _InputIterator = view::detail::TransformIterator<hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >; <template-parameter-2-2> = void; _Tp = std::__cxx11::basic_string<char>; _Alloc = std::allocator<std::__cxx11::basic_string<char> >; allocator_type = std::allocator<std::__cxx11::basic_string<char> >]'
src/utils/stl.hh:278:44:   required from 'std::vector<typename std::conditional<is_same_v<T, void>, typename std::iterator_traits<decltype (std::begin(range))>::value_type, T>::type> to_vector(Range&&) [with T = void; Range = view::detail::Transform<const hash_map<std::basic_string_view<char>, std::vector<std::basic_string_view<char> >, XXHasher>&, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >; typename std::conditional<is_same_v<T, void>, typename std::iterator_traits<decltype (std::begin(range))>::value_type, T>::type = std::__cxx11::basic_string<char>; typename std::iterator_traits<decltype (std::begin(range))>::value_type = std::__cxx11::basic_string<char>; decltype (std::begin(range)) = view::detail::TransformIterator<hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >, openmsx::printItemMap(const GroupedItems&)::<lambda(auto:62&)> >]'
  278 |         return {std::begin(range), std::end(range)};
      |                                                   ^
src/CommandLineParser.cc:423:27:   required from here
  423 |         auto printSet = to_vector(view::transform(itemMap, [](auto& p) {
      |                         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  424 |                 return strCat(formatSet(p.second, 15), ' ',
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  425 |                               formatHelpText(p.first, 50, 20));
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  426 |         }));
      |         ~~~                       
src/utils/view.hh:210:29: error: no match for 'operator-' (operand types are 'const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >' and 'const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >')
  210 |                 return x.it - y.it;
      |                        ~~~~~^~~~~~
In file included from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/stl_algobase.h:67,
                 from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/array:43,
                 from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/span:43,
                 from src/CLIOption.hh:4,
                 from src/CommandLineParser.hh:4,
                 from src/CommandLineParser.cc:1:
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/stl_iterator.h:618:5: note: candidate: 'template<class _IteratorL, class _IteratorR> constexpr decltype ((__y.base() - __x.base())) std::operator-(const reverse_iterator<_IteratorL>&, const reverse_iterator<_IteratorR>&)'
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/stl_iterator.h:618:5: note:   template argument deduction/substitution failed:
src/utils/view.hh:210:29: note:   'const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >' is not derived from 'const std::reverse_iterator<_IteratorL>'
  210 |                 return x.it - y.it;
      |                        ~~~~~^~~~~~
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/stl_iterator.h:1790:5: note: candidate: 'template<class _IteratorL, class _IteratorR> constexpr decltype ((__x.base() - __y.base())) std::operator-(const move_iterator<_IteratorL>&, const move_iterator<_IteratorR>&)'
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/stl_iterator.h:1790:5: note:   template argument deduction/substitution failed:
src/utils/view.hh:210:29: note:   'const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >::Iter<const hash_set<std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > >, hash_set_impl::ExtractFirst, XXHasher, std::equal_to<void> >, const std::pair<std::basic_string_view<char>, std::vector<std::basic_string_view<char> > > >' is not derived from 'const std::move_iterator<_IteratorL>'
  210 |                 return x.it - y.it;
      |                        ~~~~~^~~~~~
make: *** [build/main.mk:484: derived/x86_64-linux-opt/obj/CommandLineParser.cc.o] Error 1
$
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.11, NixOS, 25.05 (Warbler), 25.05.805977.88983d4b665f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/fw0ra797qv90xhr2033xbjlxv0520zaw-nixos-25.05/nixos`

### Notify maintainers

Unfortunately, `openmsx` doesn’t have any maintainers at the moment. That being said, @LordGrimmauld was the last person to touch the `openmsx` package.

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/op/openmsx/package.nix
+++ b/pkgs/by-name/op/openmsx/package.nix
@@ -1,6 +1,7 @@
 {
   lib,
   stdenv,
+  fetchpatch2,
   fetchFromGitHub,
   pkg-config,
   SDL2,
@@ -31,6 +32,14 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-iY+oZ7fHZnnEGunM4kOxOGH2Biqj2PfdLhbT8J4mYrA=";
     fetchSubmodules = true;
   };
+
+  patches = [
+    (fetchpatch2 {
+      name = "fix_view_operator.patch";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/fix_view_operator.patch?h=openmsx&id=aa63ce478c7f528d60b79bcf4c9427101caa3b94";
+      hash = "sha256-3wmUJQrM5P3zfFJt+HF32AchNSqCgFTnQ508Bztg4uA=";
+    })
+  ];
 
   nativeBuildInputs = [
     pkg-config

